rules/CVE-2021-25742/raw.rego

```diff
@@ -29,78 +37,32 @@ is_nginx_image(image) {
 	contains(image, "ingress-nginx")
 }
 
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    startswith(tag, "v")
-    tag <= "v0.49"
-}
-	
-is_vulnerable(image, namespace) {
-	contains(image, "@")
-	version := split(image, ":")
-	tag := split(version[count(version)-2], "@")[0]
-    startswith(tag, "v")
-    tag  == "v1.0.0"
-}
-
-is_vulnerable(image, namespace) {
-	not contains(image, "@")
-	version := split(image, ":")
-	tag := version[count(version)-1]
-    startswith(tag, "v")
-	tag <= "v0.49"
+is_allow_snippet_annotation_on(namespace) {
```

Comment:
I don't entirely understand the flow here. You take all ConfigMaps and filer only those in the given namespace (Q: can ingress controller use a global config map, not bound to a namespace?); then you look at the field "allow-snippet-annotations" in any configmap (not necessarily the one that belongs to the ingress controller) to see if anything is not "false"? But what happens is some configmap does not have this field at all or someone has it bit it is not nginx or an old backup configmap?

Reply:
Theoreticaly you can build an ingress controler which uses configmaps from another namespace, but the deployment mode makes this hard (ingress-controler service gets role/rolebinding is its own namespace). Parsing which configmap it uses is not easy (if possible at all)
The second point is not an issue IMHO, this has to be in the configmap